crates/polars-core/src/chunked_array/ops/sort/arg_sort.rs

```diff
@@ -105,12 +185,55 @@ pub(super) fn arg_sort_no_nulls<I, J, T>(
     iters: I,
     options: SortOptions,
     len: usize,
+    is_sorted_flag: IsSorted,
 ) -> IdxCa
 where
     I: IntoIterator<Item = J>,
     J: IntoIterator<Item = T>,
     T: TotalOrd + Send + Sync,
 {
+    // Fast path
+    // 1) If array is already sorted in the required ordered .
+    // 2) If array is reverse sorted -> we do a stable reverse.
+    if is_sorted_flag != IsSorted::Not {
+        let len_final = if let Some((limit, _desc)) = options.limit {
+            let limit = limit as usize;
```

Comment:
I think we can simplify here somewhat after we replace with `std::cmp::min`

Reply:
Have rewritten this, let me know if it looks fine now.